io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/reader/internal/wfs/WfsBackedGmlInstanceCollection.java

```diff
@@ -225,41 +227,6 @@ public WfsBackedGmlInstanceCollection(LocatableInputSupplier<? extends InputStre
 			}
 		}
 
-		// Use primordial URI and issue "hits" request to check if the WFS will
-		// return anything at all
-		int hits;
-		if (ignoreNumberMatched) {
-			hits = UNKNOWN_SIZE;
-		}
-		else {
-			try {
-				hits = requestHits(primordialUri);
-			} catch (WFSException e) {
-				log.debug(MessageFormat.format(
-						"Failed to perform hits query (REQUESTTYPE=hits): {0}", e.getMessage()), e);
-				hits = UNKNOWN_SIZE;
-			}
-		}
-
-		switch (wfsVersion) {
-		case "1.1.0":
-			// The "numberOfFeatures" reported by a 1.1.0 WFS may be smaller
-			// than the actual number of features matches by the query if the
-			// number of features returned per query is limited on the server
-			// side. Therefore do not rely on it as a size information here.
-			this.size = UNKNOWN_SIZE;
-			break;
-		case "2.0.0":
-		case "2.0.2":
-			// The "numberMatched" reported by a 2.0.0/2.0.2 WFS should be
-			// number of features matched by the query. If hits equals
-			// UNKNOWN_SIZE then size is also set to that value
-			this.size = isLimited() ? Math.min(maxNumberOfFeatures, hits) : hits;
-			break;
-		default:
-			this.size = UNKNOWN_SIZE;
-		}
-
```

Comment:
I'm not sure why this must be removed. I think this can still be used in general (as before) when the query does not contain a `RESOLVEDEPTH` parameter, or am I missing something?